src/basic/log.h

```diff
@@ -36,10 +36,11 @@ typedef enum LogTarget{
         _LOG_TARGET_INVALID = -1
 } LogTarget;
 
-#define LOG_REALM_PLUS_LEVEL(realm, level)      \
-        ((realm) << 10 | (level))
-#define LOG_REALM_REMOVE_LEVEL(realm_level)     \
-        ((realm_level >> 10))
+#define LOG_REALM_PLUS_LEVEL(realm, level)  ((realm) << 10 | (level))
+#define LOG_REALM_REMOVE_LEVEL(realm_level) ((realm_level) >> 10)
+#define SYNTHETIC_ERRNO(num)                (1u << 30 | (num))
```

Comment:
hmm, the result of this is unsigned, no? but i think it should be signed, right? hence, better `1` instead of `1u`?

also, naturally i wouldn't know whether `<<` or `|` is stronger binding, are you sure this shouldn't have `()` around the `<<` expression?

Reply:
The precedence order was explicitly designed to allow `<<` to be used to shift bits around, which are then combined using `|` and `&`. So I think it pretty clear in this case. I'll just add a comment to remind the reader. (Excessive parens make it hard to read. The old code got the parenethesizing of an arg wrong in one place, which was hard to see already. Besides, I don't like lisp.)